lib/include/ultrahdr/ultrahdr.h

```diff
@@ -111,6 +116,69 @@ struct ultrahdr_metadata_struct {
   // HDR capacity to apply the map completely
   float hdrCapacityMax;
 };
+
+/*
+ * Holds information for uncompressed image or gain map.
+ */
+struct jpegr_uncompressed_struct {
+  // Pointer to the data location.
+  void* data;
+  // Width of the gain map or the luma plane of the image in pixels.
+  size_t width;
+  // Height of the gain map or the luma plane of the image in pixels.
+  size_t height;
+  // Color gamut.
+  ultrahdr_color_gamut colorGamut;
+
+  // Values below are optional
+  // Pointer to chroma data, if it's NULL, chroma plane is considered to be immediately
+  // after the luma plane.
+  void* chroma_data = nullptr;
+  // Stride of Y plane in number of pixels. 0 indicates the member is uninitialized. If
+  // non-zero this value must be larger than or equal to luma width. If stride is
+  // uninitialized then it is assumed to be equal to luma width.
+  size_t luma_stride = 0;
+  // Stride of UV plane in number of pixels.
+  // 1. If this handle points to P010 image then this value must be larger than
+  //    or equal to luma width.
+  // 2. If this handle points to 420 image then this value must be larger than
+  //    or equal to (luma width / 2).
+  // NOTE: if chroma_data is nullptr, chroma_stride is irrelevant. Just as the way,
+  // chroma_data is derived from luma ptr, chroma stride is derived from luma stride.
+  size_t chroma_stride = 0;
+  // Pixel format.
+  uhdr_img_fmt_t pixelFormat = UHDR_IMG_FMT_UNSPECIFIED;
+  // Color range.
+  uhdr_color_range_t colorRange = UHDR_CR_UNSPECIFIED;
+};
+
+/*
+ * Holds information for compressed image or gain map.
+ */
+struct jpegr_compressed_struct {
```

Comment:
Should this also changed to "uhdr_*"

Reply:
This is not required. I have kept the legacy type names as is with out making any changes to maintain compatibility with android. Earlier, the uhdr_compressed_image_t is converted to jpegr_compressed_struct and sent to decodeJPEGR. Now the reverse happens. Through out the library this change is now using uhdr_* structs and the legacy structures are mapped to uhdr_* structs.

For example,
encode API-0
  uhdr_error_info_t encodeJPEGR(uhdr_raw_image_t* hdr_intent, uhdr_compressed_image_t* dest,
                                int quality, uhdr_mem_block_t* exif) { ... } 
is the current implementation.

 // legacy implementation of encode API - 0
  status_t encodeJPEGR(jr_uncompressed_ptr p010_image_ptr, ultrahdr_transfer_function hdr_tf,
                       jr_compressed_ptr dest, int quality, jr_exif_ptr exif) { ... } is the alias of the above. 

This function maps its arguments to uhdr_* and calls current implementation.

To summarize, ultrahdr.h contains all the legacy structures and are marked for deprecation.